DevelopGitHub/GitHubExtension/GitHubExtention.Preferences.WebApi/Package/PreferencesPackage.cs

```diff
@@ -0,0 +1,42 @@
+﻿using GitHubExtention.Preferences.WebApi.Constants;
+using GitHubExtention.Preferences.WebApi.Queries;
+using Microsoft.WindowsAzure;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using SimpleInjector;
+using SimpleInjector.Packaging;
+
+namespace GitHubExtention.Preferences.WebApi.Packages
+{
+    public class PreferencesPackage : IPackage
+    {
+        public void RegisterServices(Container container)
+        {
+            container.Register<CloudBlobContainer>(
+                () =>
+            {
+                string str = CloudConfigurationManager.GetSetting(AvatarConstants.StorageConnectionString);
+                var account = CloudStorageAccount.Parse(str);
+                var client = account.CreateCloudBlobClient();
+                var azureContainer = client.GetContainerReference(AvatarConstants.AzureContainer);
+                azureContainer.CreateIfNotExists();
+                var permissions = azureContainer.GetPermissions();
+                if (permissions.PublicAccess == BlobContainerPublicAccessType.Off)
+                {
+                    permissions.PublicAccess = BlobContainerPublicAccessType.Blob;
+                    azureContainer.SetPermissions(permissions);
+                }
+
+                return azureContainer;
+            },
+            Lifestyle.Singleton);
+            container.Register<IAzureContainerQuery, AzureContainerQuery>();
+            container.Register<AzureBlobStorageMultipartProvider>(
```

Comment:
I'm not sure exectly, I guess if user not null he has permissions to update photo. I check permissions because they can be changed in the azure out of code. So, if supose that permissions won't be changed in Azure this checking have no sence. Do I need get it away?